File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function(customers) {
    let min_penalty = 0;
    let cur_penalty=0;
    let earlier_hour =0;
    
    for(let i=0; i<customers.length;i++){
        if(customers.charAt(i)=='Y'){
            cur_penalty--;
        }
        else {
            cur_penalty++;
        }
        
        if(cur_penalty < min_penalty){
            earlier_hour =i+1;
            min_penalty = cur_penalty;
        }
    }
    return earlier_hour;
};

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.java
class Solution {
    public int bestClosingTime(String customers) {
       int min_Penalty =0;
       for(int i=0; i<customers.length();i++){
           if(customers.charAt(i)=='Y') min_Penalty++;
       }
       int cur_Penalty =min_Penalty;
       int earlier_hour =0;
       int n = customers.length();

       for(int i=0; i<n;i++){
           if(customers.charAt(i)=='Y'){
               cur_Penalty--;
           }else {
               cur_Penalty++;
           }

           if(cur_Penalty <min_Penalty){
               earlier_hour= i+1;
               min_Penalty = cur_Penalty;
           }
       }

       return earlier_hour;
    }
}